core/src/workflow/machines/upsert_search_attributes_state_machine.rs

```diff
@@ -1,40 +1,216 @@
+    coresdk::workflow_commands::{
+        UpsertWorkflowSearchAttributes,
+        workflow_command::Variant::UpsertWorkflowSearchAttributesCommandAttributes,
+    },
+    temporal::api::{
+        command::v1::Command,
+        command::v1::command::Attributes,
+        common::v1::SearchAttributes,
+        enums::v1::{CommandType, EventType},
+        history::v1::HistoryEvent,
+    },
+};
+
+use crate::workflow::machines::{Cancellable, EventInfo, MachineKind, Machines, WFMachinesAdapter};
+
+use super::{NewMachineWithCommand, workflow_machines::{MachineResponse, WFMachinesError}};
 
 fsm! {
-    pub(super) name UpsertSearchAttributesMachine; command UpsertSearchAttributesCommand; error UpsertSearchAttributesMachineError;
+    pub(super) name UpsertSearchAttributesMachine;
+    command UpsertSearchAttributesMachineCommand;
+    error WFMachinesError;
+    shared_state SharedState;
 
-    Created --(Schedule, on_schedule) --> UpsertCommandCreated;
+    // Machine is instantiated into the Created state and then transitions into the CommandIssued
+    // state when it receives the CommandScheduled event that is a result of looping back the
+    // Command initially packaged with NewMachineWithCommand (see upsert_search_attrs)
+    Created --(CommandScheduled) --> CommandIssued;
 
-    UpsertCommandCreated --(CommandUpsertWorkflowSearchAttributes) --> UpsertCommandCreated;
-    UpsertCommandCreated --(UpsertWorkflowSearchAttributes, on_upsert_workflow_search_attributes) --> UpsertCommandRecorded;
+    // Having sent the command to the server, the machine transitions into a terminal state (Done)
+    // upon observing a history event indicating that the command has been recorded. Note that this
+    // does not imply that the command has been _executed_, only that it _will be_ executed at some
+    // point in the future.
+    CommandIssued --(CommandRecorded) --> Done;
 }
 
-#[derive(thiserror::Error, Debug)]
-pub(super) enum UpsertSearchAttributesMachineError {}
+// Instantiates an UpsertSearchAttributesMachine and packs it together with an initial command
+// to apply the provided search attribute update.
+pub(super) fn upsert_search_attrs(attribs: UpsertWorkflowSearchAttributes) -> NewMachineWithCommand {
+    let mut sm = UpsertSearchAttributesMachine::new();
+    let cmd = Command {
+        command_type: CommandType::UpsertWorkflowSearchAttributes as i32,
+        attributes: Some(attribs.clone().into()),
+    };
+    NewMachineWithCommand {
+        command: cmd,
+        machine: sm.into(),
+    }
+}
 
-pub(super) enum UpsertSearchAttributesCommand {}
+// Unused but must exist
+type SharedState = ();
 
-#[derive(Default, Clone)]
+// The state-machine-specific set of commands that are the results of state transition in the
+// UpsertSearchAttributesMachine.
+#[derive(Debug, derive_more::Display)]
+pub(super) enum UpsertSearchAttributesMachineCommand {
+    #[display(fmt = "SendUpsertCommand")]
+    SendUpsertCommand(Attributes),
```

Comment:
This could potentially contain any attribute, where it should only contain the inner type which is just `UpsertSearchAttributes` (or something like that)

Reply:
You're right but it turns out this is dead code so deleting instead